xoai-data-provider/src/main/java/io/gdcc/xoai/dataprovider/repository/RepositoryConfiguration.java

```diff
@@ -146,119 +211,240 @@ public ResumptionTokenFormat getResumptionTokenFormat() {
         return this.resumptionTokenFormat;
     }
 
-    public RepositoryConfiguration and() {
-        return this;
+    @Override
+    public boolean isMetadataAttributesEnabled() {
+        return this.enableMetadataAttributes;
     }
 
-    public RepositoryConfiguration withGranularity(Granularity granularity) {
-        this.granularity = granularity;
-        return this;
-    }
+    public static final class RepositoryConfigurationBuilder {
 
-    public RepositoryConfiguration withRepositoryName(String repositoryName) {
-        this.repositoryName = repositoryName;
-        return this;
-    }
+        /* All field below package private to access in tests */
+        final List<String> adminEmails = new ArrayList<>();
+        final List<String> descriptions = new ArrayList<>();
+        final List<String> compressions = new ArrayList<>();
 
-    public RepositoryConfiguration setAdminEmails(List<String> emails) {
-        this.adminEmails.clear();
-        this.adminEmails.addAll(emails);
-        return this;
-    }
+        Granularity granularity = Granularity.Second;
+        ResumptionTokenFormat resumptionTokenFormat =
+                new SimpleResumptionTokenFormat().withGranularity(Granularity.Second);
+        String repositoryName;
+        String baseUrl;
+        Instant earliestDate;
+        DeletedRecord deleteMethod;
+        Integer maxListIdentifiers = 100;
+        Integer maxListSets = 100;
+        Integer maxListRecords = 100;
```

Comment:
Probably a stupid question, but why do we need these default values hard-coded here? - considering that we have microprofile config defaults for these?

Reply:
This has been a hardcoded default before and is also recommended within the OAI PMH implementation guidelines, too.

Yes, Dataverse has its own defaults. But anyone looking to reuse this library might have a different opinion, etc.

I do not want to add another dependency on MPConfig to this library, so I think leaving the 100 in place is a reasonable compromise between defaults, backward compatibility and reusability.

Reply:
In my defense, I did acknowledge early on it was a stupid question. (I was literally confusing what we do in the Dataverse code base and what you do in this library - sorry).